test/common/testlib.py

```diff
@@ -1192,6 +1192,23 @@ def settle_cpu(self):
                 break
             time.sleep(3)
 
+    def sed_file(self, expr, path, apply_change_action=None):
+        '''sed a file on primary machine
+
+        This is safe for @nondestructive tests, the file will be restored during cleanup.
+
+        The optional apply_change_action will be run both after sedding and after restoring the file.
+        '''
+        m = self.machine
+        m.execute("sed -i.cockpittest '{0}' {1}".format(expr, path))
+        if apply_change_action:
+            m.execute(apply_change_action)
+
+        if self.is_nondestructive():
```

Comment:
Do we actually need to check if is non destructive? Cleaning up always cannot do any harm, even when we destroy the VM right after. But fair enough.

Reply:
It's not doing harm, but it is wasting an extra round of ssh and running IO intense commands, I'd rather avoid these if right after that we'll kill the VM anyway.